src/domain/tradingview/helper.ts

```diff
@@ -0,0 +1,120 @@
+import { getServerUrl } from "config/backend";
+import { getTokenBySymbol, getWrappedToken } from "config/tokens";
+import {
+  getChainlinkChartPricesFromGraph,
+  getChartPricesFromStats,
+  getLimitChartPricesFromStats,
+  getStablePriceData,
+  timezoneOffset,
+} from "domain/prices";
+import { CHART_PERIODS, USD_DECIMALS } from "lib/legacy";
+import { formatAmount } from "lib/numbers";
+
+export const supportedResolutions = { 5: "5m", 15: "15m", 60: "1h", 240: "4h", "1D": "1d" };
+const LAST_BAR_REFRESH_INTERVAL = 15000; // 15 seconds
+
+export function getPeriodFromResolutions(value, object = supportedResolutions) {
+  return Object.keys(object).find((key) => object[key] === value);
+}
+
+function formatTimeInBar(bar) {
+  return {
+    ...bar,
+    time: bar.time * 1000,
+  };
+}
+
+async function getTokenChartPrice(chainId, symbol, period) {
+  let prices;
+  try {
+    prices = await getChartPricesFromStats(chainId, symbol, period);
+  } catch (ex) {
+    // eslint-disable-next-line no-console
+    console.warn(ex, "Switching to graph chainlink data");
+    try {
+      prices = await getChainlinkChartPricesFromGraph(symbol, period);
+    } catch (ex2) {
+      // eslint-disable-next-line no-console
+      console.warn("getChainlinkChartPricesFromGraph failed", ex2);
+      prices = [];
+    }
+  }
+  return prices;
+}
+
+async function getCurrentPriceOfToken(chainId, symbol) {
+  try {
+    const indexPricesUrl = getServerUrl(chainId, "/prices");
+    const response = await fetch(indexPricesUrl);
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const indexPrices = await response.json();
+    let symbolInfo = getTokenBySymbol(chainId, symbol);
+    if (symbolInfo.isNative) {
+      symbolInfo = getWrappedToken(chainId);
+    }
+    return indexPrices[symbolInfo.address];
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err);
+  }
+}
+
+export async function getHistoryBars({ ticker, resolution, chainId, isStable, countBack }) {
+  const period = supportedResolutions[resolution];
+  const bars = isStable ? getStablePriceData(period, countBack) : await getTokenChartPrice(chainId, ticker, period);
+  return bars.map(formatTimeInBar);
+}
+
+const getLastBarAfterInterval = (function () {
```

Comment:
The closure looks a little dangerous here - The variables startTime, lastBar, lastTicker, lastPeriod are initiated when this file is initialized and we have no way to control / reset them

I'm not sure that this could be a practical problem right now, but in theory it could be a place that is hard to debug. (for example, if we will need to use TVChart in several places)

Maybe we could pass these variables through parameters and store them in a state or ref? Or at least create a closure in the hook and pass this function as a callback to getLiveBar